test/mcmc/particle_mcmc.jl

```diff
@@ -114,6 +127,24 @@ end
         @test length(unique(c[:s])) == 1
     end
 
+    @testset "addlogprob leads to reweighting" begin
+        # Make sure that PG takes @addlogprob! into account. It didn't use to:
+        # https://github.com/TuringLang/Turing.jl/issues/1996
+        @model function addlogprob_demo()
+            x ~ Normal(0, 1)
+            if x < 0
+                @addlogprob! -2.0
+            else
+                # Need a balanced number of addlogprobs in all branches, or
+                # else PG will error
+                @addlogprob! 0.0
+            end
+        end
+        c = sample(addlogprob_demo(), PG(10), 100)
+        # Result should be biased towards x > 0.
+        @test mean(c[:x]) > 0.5
```

Comment:
```suggestion
        @test mean(c[:x]) > 0.7
```
Could this get a bit more clearance from 0.5, with, if necessary, the `@addloprob!` value being increased? Otherwise there's 50% chance this will pass just by luck.

Reply:
Also, I don't know why I didn't put in StableRNGs here. Will do so too.